src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +471,240 @@ internal String TranslateNode(QueryNode node)
+
+            string inOperator = ExpressionConstants.KeywordIn;
+            string[] inSeparator = { inOperator };
+            string[] substrings = translatedNode.Trim().Split(inSeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftIn = substrings[0];
+            string rightIn = substrings[1];
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftIn = "$it/" + leftIn;
+                translatedNode = leftIn + inOperator + rightIn;
+            }
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="AnyNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindAnyNode(AnyNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node);
+
+            const string slashAny = "/any";
+            string[] anySeparator = { slashAny };
+            string[] substrings = translatedNode.Trim().Split(anySeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftAnyNodeSubstring = substrings[0];
+            string rightAnyNodeSubstring = substrings[1];
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftAnyNodeSubstring = "$it/" + leftAnyNodeSubstring;
+                translatedNode = leftAnyNodeSubstring + slashAny + rightAnyNodeSubstring;
+            }
+            return translatedNode;
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="SingleValueFunctionCallNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindSingleValueFunctionCallNode(SingleValueFunctionCallNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode firstParameterRangeVariableNode = GetResourceRangeVariableReferenceNode(node);
+
+            char[] separators = { '(', ')' };
+            string[] subtrings = translatedNode.Trim().Split(separators, StringSplitOptions.RemoveEmptyEntries);
+            string withinBrackets = subtrings[1];
+            string[] parameters = withinBrackets.Trim().Split(',');

```

Comment:
Are spaces possible in the parameters string, i.e. `(a, b, c)` as opposed to `(a,b,c)`, and should we bother trimming them?

Reply:
Good catch!

Reply:
#Resolved